test/validate.spec.js

```diff
@@ -1,42 +1,43 @@
 /**
  * Basic spec for lib/validate.js
- *
- * N.b., I stub all the crypto stuff here because I have no idea how to generate a test token. -Æ.
  */
 
+const NodeRSA = require('node-rsa');
+const crypto = require('crypto');
 const chai = require('chai');
 const sinon = require('sinon');
 const sinonChai = require('sinon-chai');
-const proxyquire = require('proxyquire');
+const validatorFactory = require('../validate');
 
 chai.should();
 chai.use(sinonChai);
 
-// Stubs for node-rsa
-const rsaStub = sinon.stub();
-const exportKeyStub = sinon.stub();
-rsaStub.returns({ exportKey: exportKeyStub });
-exportKeyStub.returns('test-pem');
-
-// Stubs for crypto
-const createVerifyStub = sinon.stub();
-const verifyStub = sinon.stub();
-const updateSpy = sinon.spy();
-createVerifyStub.returns({ update: updateSpy, verify: verifyStub });
-verifyStub.returns(true);
-
-const validatorFactory = proxyquire('../validate', {
-	'node-rsa': rsaStub,
-	'crypto': {
-		createVerify: createVerifyStub,
-	},
-});
+// s3o uses rsa to generate key pairs
+// https://github.com/Financial-Times/s3o/blob/992be3a122006210013da64aa2ceee65e03a479d/src/main/java/com/ft/s3o/util/KeyGenerator.java#L19
+const generateRSAKeyPair = () => new NodeRSA().generateKeyPair(512);
+
+const sign = (data, key) => {
+	const privatePem = key.exportKey('pkcs8-private-pem');
+	const s = crypto.createSign('sha1');
+	s.update(data);
+	s.end();
+	return s.sign(privatePem);
+}
 
 const keyFactory = sinon.stub();
 
 describe('lib/validate.js', () => {
+	let publicKey
+	let key
```

Comment:
is this variable used anywhere?

Reply:
yup. `publicKey` is used to setup stubs. `key` is used to in each `sign` call.

Renamed to `keyPair` for clarity.